analysis/02_trigger_2022_iri.md

```diff
@@ -0,0 +1,382 @@
+# Evaluate trigger status during 2022 season
+
+During 2022 the trigger was not reached. As evaluation the country team asked
+to provide some figures which are computed here
+
+The trigger is evaluated at two points in time:
+
+- Trigger I: In March, if the trigger is met for risk  of below average
+precipitation in the June, July, August period5,
+funds  are  released  for  the  commencement  of  activities  by  all
+selected  sectors,  including  food  security  and
+livelihoods, nutrition, protection, as well as water, hygiene, and sanitation.
+- Trigger II: In July, if the trigger is met for risk of below average
+precipitation in the August, September, and October
+period, funds are released for the implementation of activities by the
+food security and livelihoods sector.
+
+The trigger is defined as:
+
+1. 40% or more probability of below average rainfall AND
+2. The probability of below average rainfall should be 5 percentage points
+higher than that of above average rainfall.
+
+Finally,  activation  is  also  contingent  on  at  least  10%  of  the  zone
+(Boucle  de  Mouhoun,  Centre  Nord,  Sahel,  and  Nord)
+meeting these criteria. This criterion aims to capture through geographical
+spread a more severe e shock,  in line with the
+humanitarian objectives of the pilot. Overall, this framework aims to
+target 1 in 4-year type of event
+
+```python
+%load_ext autoreload
+%autoreload 2
+%load_ext jupyter_black
+```
+
+```python
+import geopandas as gpd
+import pandas as pd
+import rioxarray
+import numpy as np
+import xarray as xr
+import cftime
+import calendar
+from dateutil.relativedelta import relativedelta
+from matplotlib.colors import ListedColormap
+from rasterio.enums import Resampling
+import hvplot.xarray
+import altair as alt
+import matplotlib.pyplot as plt
+
+import aatoolbox.utils.raster
+from aatoolbox import (
+    CodAB,
+    IriForecastProb,
+    IriForecastDominant,
+    GeoBoundingBox,
+)
+
+from src import utils, constants
+```
+
+## Set variables
+
+```python
+# Get geometry parameters
+codab = CodAB(country_config=constants.country_config)
+gdf_adm1 = codab.load(admin_level=1)
+geobb = GeoBoundingBox.from_shape(gdf_adm1)
+gdf_aoi = gdf_adm1[gdf_adm1.ADM1_FR.isin(constants.adm_sel)]
+```
+
+```python
+trig_year = 2022
+```
+
+## Inspect forecasts
+
+We load the iri data indicating the dominant tercile.
+The negative values indicate forecasted below average rainfall,
+and the positive values above average. We use rounded bins (e.g. -40 to 40).
+The IRI website bins are slightly different,
+where values between -37.5 and 37.5 are assigned to the normal tercile.
```

Comment:
Good place to add a note about the legend colours of the forecast not matching the actual colours in the map, for the normal tercile?